### ISSUE TITLE
Accept browser URL in install_github()

### ISSUE THREAD

**Issue by jennybc** [r-lib/remotes#52]
Is this where the functionality of `devtools::install_github()` will land after the diaspora is complete? Or should I open this there?

It would be really nice if a browser URL worked in `install_github()`.

``` r
remotes::install_github("https://github.com/juliasilge/southafricastats")
#> Error in username %||% getOption("github.user") %||% stop("Unknown username."): Unknown username.
```

vs.

``` r
remotes::install_github("juliasilge/southafricastats")
#> Using github PAT from envvar GITHUB_PAT
#> Downloading GitHub repo juliasilge/southafricastats@master
#> Installing package into '/Users/jenny/resources/R/library'
#> (as 'lib' is unspecified)
```

**Comment by jennybc** [comment]
Last time I checked, this is the regex built in `parse_github_repo_spec()`. It is clearly crying out to be extended 😜

```
^(?:(?:([^/]+)/)?([^/@#]+)(?:/([^@#]*[^@#/]))?(?:(?:@([^*].*))|(?:#([0-9]+))|(?:@([*]release)))?|(.*))$
```

**Comment by gaborcsardi** [comment]
An easy workaround is `install_git`:

```r
remotes::install_git("https://github.com/juliasilge/southafricastats")
```

**Comment by gaborcsardi** [comment]
Or you want it to work from any page on GitHub, correctly handling branches, commits and PRs? That would be a cool feature, actually. 

But I would not extend the regexp, I would just create a new one. The intersection of the two types of strings is empty, so no need for a single regex.

**Comment by wibeasley** [comment]
@jennybc, should this issue be closed?  It seems like you fixed it 8 months later in ef0eb6663e1d9b6d74cfdd5e56fc434b060ff0d9.  If so, this issue might be a duplicate of https://github.com/r-lib/remotes/issues/90.

**Comment by jennybc** [comment]
Yes looks like it should be closed.
